src/app/grid/grid-validator-service-extended/grid-validator-service-extended.component.ts

```diff
@@ -55,46 +53,37 @@ export class GridValidatorServiceExtendedComponent {
      * Bind this handler to `cellEdit` output of the grid
      * in order to cancel cell editing in case the submitted
      * value is invalid.
-     * 
-     * @param evt 
+     *
+     * @param evt
      */
+
     public cellEdit(evt) {
-        if (!evt.isValid) {
+        if (!evt.valid) {
             evt.cancel = true;
         }
     }
 
     public commit() {
-        const invalidTransactions = this.grid.transactions.getInvalidTransactionLog();
+        const invalidTransactions = this.grid.validation.getInvalid();
         if (invalidTransactions.length > 0) {
             if (confirm('You\'re commiting invalid transactions. Are you sure?')) {
-                this.grid.transactions.commit(this.transactionData);
+                this.grid.validation.clear();
             }
         } else {
-            this.grid.transactions.commit(this.transactionData);
+            this.grid.validation.clear();
         }
     }
 
-    private _testFax(): ValidatorFn {
-        return (control: AbstractControl): ValidationErrors | null => {
-            const match = this._regexValidationFunction(control.value, /\+\d{1}\-(?!0)(\d{3})\-(\d{3})\-(\d{4})\b/i);
-            return !match ? { forbiddenFax: { value: control.value } } : null;
-        }
-    }
-
-    private _regexValidationFunction(value: any, rgx: RegExp): boolean {
-        const match = rgx?.test(value);
-        return !!match;
-    }
-
-    private _testDateRecord(thresholdVal?: any): ValidatorFn {
+    public _testDateRecord(thresholdVal?: any): ValidatorFn {
```

Comment:
Please remove the thresholdVal parameter. It gets passed only once when the form group is created for given record and will never be updated for the same record, so it's no reliable to use it. If you want to compare the lastActiveRecord value with the createdOnRecord value you can get them from the formGroup for exampe: https://github.com/IgniteUI/igniteui-angular-samples/pull/3073/commits/22fcc61d4f73def909dda11be166489555ac16d0#r955045801

Reply:
Also the name of the validator is not very descriptive. Maybe something like datesThresholdValidator would be better.